source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/organizations/settings/select-logo/index.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+
+export interface IProps {
+  onChange: (imageData: String) => void;
+  value?: string;
+}
+
+export interface IState {
+  imageData: String;
+}
+
+class SelectLogo extends React.Component<IProps, IState> {
+  state = { imageData: null };
+
+  componentWillReceiveProps(nextProps) {
+    const { value } = nextProps;
+
+    if (value) {
+      this.setState({ imageData: value });
+    }
+  }
+
+  handleNewImage = e => {
```

Comment:
we should refactor this code into a helper

Reply:
agreed. I'll do that in my next PR, since this one is so huge.